provisioning_client/tests/common_prov_e2e/prov_hsm/x509_msr.c

```diff
@@ -161,20 +149,21 @@ static int generate_root_ca_info(X509_CERT_INFO* x509_info, RIOT_ECC_SIGNATURE*
             result = 0;
         }
     }
+    ecc_signature_destroy(&tbs_sig);
     return result;
 }
 
-static int produce_device_cert(X509_CERT_INFO* x509_info, RIOT_ECC_SIGNATURE tbs_sig)
+static int produce_device_cert(X509_CERT_INFO* x509_info)
 {
     int result;
     uint8_t der_buffer[DER_MAX_TBS] = { 0 };
     DERBuilderContext der_ctx = { 0 };
     RIOT_STATUS status;
-
+    RIOT_ECC_SIGNATURE tbs_sig = { 0 };
 
     // Build the TBS (to be signed) region of DeviceID Certificate
     DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
-    if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &x509_info->device_id_pub) != 0)
+    if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &x509_info->device_id_pub, (uint8_t*)&x509_info->device_id_pub, sizeof(x509_info->device_id_pub)) != 0)
```

Comment:
Why isn't the root pub portion used here (similar to the production code implementation)?

Reply:
Not sure I understand
```C
if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub, (uint8_t*)&riot_info->device_id_pub, sizeof(riot_info->device_id_pub)) != 0)
```
This is from produce_device_cert hsm_client_riot.c ln: 217
the optional params are the same, using the devic_id_pub, should the TBS call all use the same key info?

Reply:
See https://github.com/microsoft/RIoT/blob/master/Reference/RIoT/RIoTCrypt/include/x509bldr.h#L40:

```C
int
X509GetDeviceCertTBS(
    DERBuilderContext   *Tbs,
    RIOT_X509_TBS_DATA  *TbsData,
	RIOT_ECC_PUBLIC     *DevIdKeyPub,
	uint8_t             *RootKeyPub,
	uint32_t             RootKeyPubLen
);```

The high-level idea for RIoT is to produce an X509 cert chain rooted in the root-of-trust certificate. Somehow the elements need to be linked together.

@DMattoon PTAL

Reply:
If self-signed, then the "root" parameters are not meaningful.  If not self-signed then the root parameters are recommended but not required.

Reply:
Fixed